infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/transport/sample/GRPCLogReplicationServerChannelAdapter.java

```diff
@@ -36,7 +37,8 @@ public GRPCLogReplicationServerChannelAdapter(ServerContext serverContext, LogRe
         super(serverContext, router);
         this.service = new GRPCLogReplicationServerHandler(router);
         this.port = Integer.parseInt((String) serverContext.getServerConfig().get("<port>"));
-        this.server = ServerBuilder.forPort(port).addService(service).build();
+        this.server = ServerBuilder.forPort(port).addService(service)
+                .executor(Executors.newSingleThreadScheduledExecutor()).build();
```

Comment:
Trivial question. Just for my knowledge. Can you please describe the reason on why do we need single threaded server?

Reply:
https://github.com/CorfuDB/CorfuDB/blob/003d656cfe17f35a3d6814bd9b31e1c16e679b6b/infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/SnapshotSender.java#L144-L157
Sender side expects ACKs to be received in order, otherwise exception will be thrown. Actually in our netty implementation for rpc handlers, the executor is single threaded:
https://github.com/CorfuDB/CorfuDB/blob/003d656cfe17f35a3d6814bd9b31e1c16e679b6b/infrastructure/src/main/java/org/corfudb/infrastructure/LogReplicationServer.java#L77-L84
The gRPC implemented handlers are only used in tests, previously it used default thread pool executor, which could lead to ACK mismatch at sender side.